routing/payment_lifecycle.go

```diff
@@ -144,19 +148,20 @@ func (p *paymentLifecycle) resumePayment() ([32]byte, *route.Route, error) {
 
 				// If there was an error already recorded for this
 				// payment, we'll return that.
-				if p.lastError != nil {
-					return [32]byte{}, nil, errNoRoute{lastError: p.lastError}
+				if shardHandler.lastError != nil {
+					return [32]byte{}, nil, errNoRoute{
+						lastError: shardHandler.lastError,
+					}
```

Comment:
This is removed again in a later commit. Ideally the commits in a pr don't rewrite the same code.

Reply:
It is a tradeoff between huge commits and what you mention. I opted for smaller commits to ensure tests pass all the way trhough the PR